### ISSUE TITLE
[DNSClient] Error while using lookup

### ISSUE THREAD

**Issue by phiz71** [eclipse-vertx/vert.x#5035]
### Version

Tested in all versions from 3.9.x to 4.5.x

### Context

I encountered an exception which looks suspicious while using a simple lookup call to a public DNS using the DnsClient.

With this configuration:
```java
DnsClient client = Vertx.vertx().createDnsClient(53, "212.27.40.240"); // Public DNS
```
When I execute a lookup with the DnsClient:
```java
client.lookup("google.com")
```
Then I get this error:
```
io.vertx.core.VertxException: DNS query timeout for google.com.
	at io.vertx.core.dns.impl.DnsClientImpl$Query.lambda$null$0(DnsClientImpl.java:352)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:264)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:246)
	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:43)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

By activating DEBUG log on netty's classes, I get this:
```
11:36:59.222 [vert.x-eventloop-thread-0] [] DEBUG i.n.util.ResourceLeakDetectorFactory - Loaded default ResourceLeakDetector: io.netty.util.ResourceLeakDetector@51a297a1
11:36:59.228 [vert.x-eventloop-thread-0] [] DEBUG i.n.handler.logging.LoggingHandler - [id: 0xb1852f87] WRITE: DatagramPacket(=> /212.27.40.240:53, PooledUnsafeDirectByteBuf(ridx: 0, widx: 44, cap: 1024)), 44B
11:36:59.228 [vert.x-eventloop-thread-0] [] DEBUG i.n.handler.logging.LoggingHandler - [id: 0xb1852f87] FLUSH
io.vertx.core.VertxException: DNS query timeout for google.com.
	at io.vertx.core.dns.impl.DnsClientImpl$Query.lambda$null$0(DnsClientImpl.java:352)
```

#### Note
If I use the Google DNS (8.8.8.8), everything works:
```
11:41:38.984 [vert.x-eventloop-thread-0] [] DEBUG i.n.util.ResourceLeakDetectorFactory - Loaded default ResourceLeakDetector: io.netty.util.ResourceLeakDetector@2de81aaa
11:41:38.990 [vert.x-eventloop-thread-0] [] DEBUG i.n.handler.logging.LoggingHandler - [id: 0xa830586a] WRITE: DatagramPacket(=> /8.8.8.8:53, PooledUnsafeDirectByteBuf(ridx: 0, widx: 44, cap: 1024)), 44B
11:41:38.991 [vert.x-eventloop-thread-0] [] DEBUG i.n.handler.logging.LoggingHandler - [id: 0xa830586a] FLUSH
11:41:39.019 [vert.x-eventloop-thread-0] [] DEBUG i.n.handler.logging.LoggingHandler - [id: 0xa830586a] READ: DatagramPacket(/8.8.8.8:53 => /0.0.0.0:58067, PooledUnsafeDirectByteBuf(ridx: 0, widx: 44, cap: 2048)), 44B
IP address is 142.250.74.238
11:41:39.023 [vert.x-eventloop-thread-0] [] DEBUG i.n.handler.logging.LoggingHandler - [id: 0xa830586a] READ COMPLETE
```

Increasing the timeout as no effect.

### Do you have a reproducer?

https://github.com/phiz71/DebugDnsClient

### Extra

I'm using OpenJDK 64-Bit Server VM Temurin-17.0.7+7

**Comment by vietj** [comment]
can you try with Vert.x 5 snapshot ? it uses a different implementation based on the Netty resolver ?

have you tried to monitor the UDP traffic to see if the dns server provides a response ?

**Comment by phiz71** [comment]
Well I have to apologize 🙇🏻‍♂️ It seems that the issue comes from the public DNS I used for my tests. Indeed for some of them, I was not able to reach them from my laptop. But for others, the requests were OK.

But FYI, I've tried with v5-SNAPSHOT of Vert.X and it was a little more verbose (with DEBUG level)
```
10:16:16.548 [main] [] DEBUG i.n.b.ChannelInitializerExtensions - -Dio.netty.bootstrap.extensions: null
10:16:16.557 [vert.x-eventloop-thread-0] [] DEBUG i.netty.resolver.dns.DnsQueryContext - -Dio.netty.resolver.dns.idReuseOnTimeoutDelayMillis: 10000
10:16:16.559 [vert.x-eventloop-thread-0] [] DEBUG i.n.util.ResourceLeakDetectorFactory - Loaded default ResourceLeakDetector: io.netty.util.ResourceLeakDetector@22851ab
10:16:16.562 [vert.x-eventloop-thread-0] [] DEBUG i.netty.resolver.dns.DnsQueryContext - [id: 0xbea9d2a6] WRITE: UDP, [6612: /212.27.40.240:53], DefaultDnsQuestion(google.com. IN A)
10:16:16.564 [vert.x-eventloop-thread-0] [] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.maxCapacityPerThread: 4096
10:16:16.564 [vert.x-eventloop-thread-0] [] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.ratio: 8
10:16:16.564 [vert.x-eventloop-thread-0] [] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.chunkSize: 32
10:16:16.564 [vert.x-eventloop-thread-0] [] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.blocking: false
10:16:16.564 [vert.x-eventloop-thread-0] [] DEBUG io.netty.util.Recycler - -Dio.netty.recycler.batchFastThreadLocalOnly: true
10:16:16.566 [vert.x-eventloop-thread-0] [] DEBUG io.netty.buffer.AbstractByteBuf - -Dio.netty.buffer.checkAccessible: true
10:16:16.566 [vert.x-eventloop-thread-0] [] DEBUG io.netty.buffer.AbstractByteBuf - -Dio.netty.buffer.checkBounds: true
10:16:16.566 [vert.x-eventloop-thread-0] [] DEBUG i.n.util.ResourceLeakDetectorFactory - Loaded default ResourceLeakDetector: io.netty.util.ResourceLeakDetector@200e4235
10:16:16.570 [vert.x-eventloop-thread-0] [] DEBUG i.netty.resolver.dns.DnsQueryContext - [id: 0xbea9d2a6] WRITE: UDP, [36413: /212.27.40.240:53], DefaultDnsQuestion(google.com. IN AAAA)
io.netty.resolver.dns.DnsResolveContext$SearchDomainUnknownHostException: Failed to resolve 'google.com' [A(1), AAAA(28)] and search domain query for configured domains failed as well: [lan]
	at io.netty.resolver.dns.DnsResolveContext.finishResolve(DnsResolveContext.java:1097)
[...]
```

**Comment by phiz71** [comment]
Hello @vietj,
Finally I've managed to find what was my real problem.

I use `DnsClient.lookup()` to retrieve the IP adresses of a domain.
And this `DnsClient` prepares a datagram with 2 questions:
<img width="516" alt="image" src="https://github.com/eclipse-vertx/vert.x/assets/13161768/2c0e3175-d71e-481e-bd92-730d96e6d9b1">

This kind of requests is supported by Google DNS 8.8.8.8, but not all DNS Servers are able to handle such requests.

I've also tested with a `nslookup` command, and it appears that it generates two requests to the DNS Server, one by RecordType (A & AAAA)

My workaround for that is to only use `lookup4`, to have only 1 question by DNS request.

I have tested with vertx 5 and it seems that the new behaviour is to have one question by request.
